### PR TITLE
Add new symbol messages

### PR DESCRIPTION
**Pull request by eric-wieser**
This corresponds to https://github.com/leanprover-community/lean/pull/724

### CODE REVIEW

File: lean-client-js-core/src/commands.ts [hunk 1]
```diff
@@ -139,6 +139,31 @@ export interface InfoResponse extends CommandResponse {
...
+}
+
+export declare type SymbolKind
+    = 'constructor'
+    | 'field'
+    | 'class'
+    | 'inductive'
+    | 'instance'
+    | 'def'
+    | 'theorem'
```
Comment (eric-wieser) [inline lean-client-js-core/src/commands.ts:154]: One question that came up in discussion with @Julian; where should the translation from lean concepts to VScode/LSP concepts go; in the lean server, or in the extension?

Comment (Julian) [reply lean-client-js-core/src/commands.ts:154]: The motivation for the leading question was this is going to immediately need to be copied in `lean-language-server` if it doesn't go further upstream than that, but there's presumably 3 choices -- server, extension (and then copy pasted), or in some JS library function here.

Comment (gebner) [reply lean-client-js-core/src/commands.ts:154]: I'd keep the server simple if possible.  Having the raw data could also be useful for other applications.  Not that I expect a lot of new development in Lean 3 though.

In this case, copying the vscode implementation into `lean-language-server` would be my preferred solution.

**Comment by eric-wieser** [comment]
bors merge

**Comment by eric-wieser** [comment]
I suspect I don't have permission to merge this myself

**Comment by gebner** [comment]
There's no bors here! :smile:

**Pull request merged**